proposals/lip-Introduce_cross_chain_messages.md

```diff
@@ -0,0 +1,635 @@
+
+To achieve some of its basic functionalities, the Interoperability module uses three standard messages. 
+The first one, the _cross-chain update receipt_, is sent back to the partner chain (the chain with which the current chain is exchanging messages) whenever a cross-chain update transaction from this chain gets included. 
+The cross-chain update receipt serves to update a few properties of the interoperability store, and in general to acknowledge the inclusion of cross-chain update transactions. 
+The second one, the _channel terminated message_, is sent to chains which have been terminated. 
+This message also updates a few properties of the interoperability store to keep them synchronized between chains. 
+The last one, the registration message, is used when registering a chain on the Lisk mainchain. 
+The message serves as a guarantee that the correct chain ID and network ID are used when the registration transaction is sent on the sidechain.
+
+Further motivation and rationale behind the Lisk interoperability architecture is given in the general interoperability document ["Introduce Interoperability module"][base-interoperability-LIP].
+
+
+## Rationale
+
+Messages in the Lisk ecosystem will be included in multiple chains. 
+It is therefore important that all chains can deserialize and process the base part of cross-chain messages in the same way. 
+The properties of the message contain all the information necessary for this purpose while trying to keep a minimal size.
+
+
+### Cross-chain Message Properties
+
+In the following, we list all the properties of a cross-chain message.
+
+
+#### Sending Chain ID and Receiving Chain ID
+
+Used to identify the chains exchanging the cross-chain message. 
+On the mainchain the receiving chain ID is read to route the message to the corresponding chain outbox, as specified in [LIP "Introduce cross-chain update mechanism"][CCU-LIP]. 
+The sending chain ID is used for example if the message triggers an error and has to be sent back.
+
+
+#### Index
+
+When a cross-chain message is created and added to the partner chain outbox, the size of the outbox at that point is added to the message in the `index` property. 
+This allows all messages to be distinct and to be tracked throughout the ecosystem.
+
+
+#### Module ID and Cross-chain Command ID
+
+Once the message has reached the recipient chain, the two properties `moduleID` and `crossChainCommandID` specify which logic should be used to validate and execute the message. 
+The Interoperability module handles the message in case the required logic is not available on the chain. 
+This brings the benefit that sending chains do not need to monitor all other chains in the ecosystem and the modules they support. 
+We are following an optimistic approach in which all messages will be delivered and error handling will kick in when it is needed. 
+In that regard, users should be aware that valid messages will usually not trigger a response. 
```

Comment:
Yes, this is up to the module. For all the modules we have specified so far, we don't have a return message when the CCM is successful. In that regard, I think it's fair to warn users that valid messages will usually not trigger a response. What do you think should be written differently?

Reply:
Once suggestion.

> Once the message has reached the recipient chain, the two properties `moduleID` and `crossChainCommandID` specify which logic should be used to validate and execute the message. 
The Interoperability module perfoms some basic validations for all messages and then forwards the cross-chain message to the respective module which is responsible for the validation and execution. In case the required logic is not available on the chain, the Inoperability module sends a message with an error status back to the recipient chain provided the message included enough fees.
This approach brings the benefit that sending chains do not need to monitor all other chains in the ecosystem and the modules they support. 

> Apart from the basic validations and error handling, the Interoperability module does not intervene in the cross-chain messaging between modules and is agnostic of the message protocol that modules use to communicate with each other. In particular, the messaging protocol between modules can choose whether the reception and successful execution of a message is confirmed with a return message. 
For the Token and NFT module,  we are following an optimistic approach in which we assume that all messages are successfully delivered unless an error message is returned. In this case, users should be aware that valid messages send from a chain will not trigger a response message to the chain.

Reply:
The second paragraph could also be in a separate section about error handling.

Reply:
modified the `moduleID` paragraph and added some information to the `message tracking` section.

Reply:
Sounds good.